src/3_PB/SpecificheTecniche/content.typ

```diff
@@ -156,120 +157,194 @@ Per illustrare il funzionamento del sistema, abbiamo utilizzato un diagramma di
+Anche la classe _Writer_ realizza il design pattern _Strategy_. Sono state progettate due strategie, la prima, (_KafkaWriter_), atta a permettere al simulatore di inviare i messaggi contenenti i dati della rilevazione a #glossary("Kafka"), mentre la seconda atta a permettere al simulatore di stampare i risultati su terminale al fine di poterne testare il comportamento. Inoltre l'applicazione di tale #glossary[design pattern] potrebbe consentire di realizzare il componente di scrittura anche per eventuali broker dati alternativi, nel momento in cui ce ne sia il bisogno.
+Nello specifico, la classe _KafkaWriter_ realizza il suo scopo tramite l'impiego del #glossary[design pattern] _Adapter_, nella sua variante _Object Adapter_. Viene infatti fatto utilizzo della classe _Producer_ della liberia _confluent_kafka_. Dato che tale classe potrebbe essere soggetta a variazioni non controllabili da noi, si è deciso di utilizzare tale pattern per permettere di rispondere prontamente a tali cambiamenti, spostandone la complessità proprio nell'adapter.
 
 === Classi: metodi e attributi
 Si procede alla descrizione  di classi e interfacce progettate dal team.
-Non vengono menzionati i costruttori. 
-==== SimulatorExecutorFactory (Classe)
-//TODO: da rivedere perchè dobbiamo descrivere anche come funziona nel caso in cui la porta e l'host non devono essere passate (tipo per l'StdOutWriter ad esempio).
-//Secondo me (Kara) qui potremmo utilizzare un'altro strategy.
+Non vengono menzionati i costruttori.
+==== SimulatorExecutorFactoryTemplate (Classe)
 ===== Attributi
-- *config_file_name: String [Privato]*: Percorso relativo del file di configurazione dei simulatori; 
-- *data_broker_host: String [Privato]*: indirizzo ip della macchina che ospita il message broker; 
-- *data_broker_port: String [Privato]*: porta della macchina che ospita il message broker; 
-- *io_module: FileIO [Privato]*: modulo per interagire con il file system.
+- *configs: String [Protetto]*: contenuto del file di configurazione dei simulatori;
+- *simulators: SimulatorThread[\*] [Protetto]*: collezione di oggetti _SimulatorThread_.
 ===== Metodi
-- *create(): SimulatorExecutor [Pubblico]*: a partire dal file che viene passato tramite costruttore si occupa di costruire un oggetto della classe _SimulatorExecutor_.
+- *create_simulator(in config: {\*}, in simulator_type: SensorTypes, in cls: SensorSimulatorStrategy): void [Protetto]*: metodo astratto, implementato dalle classi _KafkaSimulatorExecutorFactory_ e _StdoutSimulatorExecutorFactory_;
```

Comment:
Veramente la creazione di dei dictionary si scrive in questo modo? Mi riferisco a {*}